### PR TITLE
refactored

### PR DESCRIPTION
**Pull request by mklishevych**

### CODE REVIEW

**Comment by mklishevych** [comment]
@knopa ну давай меня хотя бы попесочим за мой рефакторинг) а то остальные огорчают

**Comment by mklishevych** [comment]
ну нет и нет.

File: fork/mklishevych/actions.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
+export const successAction = (text) => {
```
Comment (knopa) [inline fork/mklishevych/actions.js:1]: is there really big difference between actions type?

File: fork/mklishevych/actions.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        callNotification: true,
+        text: text,
+        id: new Date().getTime(),
+        level: 'error'
+    };
+};
+
+export const errorByTypeAction = (type, error) => {
+    switch (type) {
+        case 'SIGNIN_USER':
```
Comment (knopa) [inline fork/mklishevych/actions.js:23]: why not aggregate errors into one return?

**Comment by knopa** [comment]
@mklishevych In general is very good. Without touching other parts of the system I don't know what we can improve here.

File: fork/mklishevych/actions.js (hunk 2 above)
Comment (mklishevych) [reply fork/mklishevych/actions.js:23]: we can, but we still have to map error type that we recieve from the server to the error type that is used in the app. so we can change this to something like this:
```javascript
const map = { SIGNIN_USER: 'ERRORS_SIGNIN_USER', ... }
return map[type] ? { type: map[type], message: error } : undefined;
```

File: fork/mklishevych/actions.js (hunk 1 above)
Comment (mklishevych) [reply fork/mklishevych/actions.js:1]: no, just level :) so yes, further refactoring might be extracting this to a function `notificationAction(text, level)` and after that `successAction` and `errorAction` would look like this:
```javascript
export const successAction(text) => notificationAction(text, 'success');
export const errorAction(text) => notificationAction(text, 'error');
```

File: fork/mklishevych/actions.js (hunk 2 above)
Comment (knopa) [reply fork/mklishevych/actions.js:23]: well as I understand that low level shouldn't know nothing about action type and error handling

File: fork/mklishevych/actions.js (hunk 1 above)
Comment (knopa) [reply fork/mklishevych/actions.js:1]: that looks better and know small question what is success and error string for?

Comment (mklishevych) [reply fork/mklishevych/actions.js:1]: I don't have an answer on that :) I made my code backward compatible. I don't know how action payload is actually used in the app. Maybe the app behaves differently depending on the `level` when the action of type `ADD_NOTIFICATION_MIDDLEWARE ` is dispatched.

File: fork/mklishevych/actions.js (hunk 2 above)
Comment (mklishevych) [reply fork/mklishevych/actions.js:23]: again, not sure how the app uses dispatched payload, so I just did it backward compatible

Comment (mklishevych) [reply fork/mklishevych/actions.js:23]: Ah, yes, I think you're right. This is low level code and it should just dispatch the generalized payload:
```javascript
export const errorByTypeAction = (type, error) => {
    return {
        type: 'RESPONSE_ERROR',
        errorType: type,
        error: error
    }
};
```
And then the old logic should be taken care of in the reducers based on errorType field

**Pull request merged**